addons/mail/static/src/components/composer/composer.js

```diff
@@ -188,10 +189,17 @@ class Composer extends Component {
             }
             return;
         }
-        await this.composer.postMessage();
-        // TODO: we might need to remove trigger and use the store to wait for the post rpc to be done
-        // task-2252858
-        this.trigger('o-message-posted');
+        try {
+            await this.composer.postMessage();
+            // TODO: we might need to remove trigger and use the store to wait for the post rpc to be done
+            // task-2252858
+            this.trigger('o-message-posted');
+        }
+        catch(e){
+            if (!e instanceof AlreadyPostingError) {
+                throw e;
+            }
+        }
```

Comment:
I think diff would be easier if we didn't call `postMessage` if `isPostingMessage` is true. Similar to the above condition in the current method, that also does not call `postMessage` in some situation.

This would remove the need to introduce this new Error and keep the model method cleaner by avoiding the preventive throw at the start.